File: app/src/main/java/com/narsha/wave_android/view/adapter/search/StateFragment.java
package com.narsha.wave_android.view.adapter.search;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.narsha.wave_android.view.fragment.search.searched.AllSearchedFragment;
import com.narsha.wave_android.view.fragment.search.searched.ArtistSearchedFragment;
import com.narsha.wave_android.view.fragment.search.searched.MusicSearchedFragment;

public class StateFragment extends FragmentStateAdapter {
    public StateFragment(@NonNull Fragment fragment) {
        super(fragment);
    }
    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment fragment;
        if(position==0){
                fragment = new AllSearchedFragment();
        } else if(position == 1) {
                fragment = new MusicSearchedFragment();
        }else{
                fragment = new ArtistSearchedFragment();
        }

        return fragment;
    }
    @Override
    public int getItemCount() {
        return 3;
    }
}

File: Wave_V2/app/src/main/java/com/project/wave_v2/widget/StateFragment.kt
package com.project.wave_v2.widget

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.project.wave_v2.view.fragment.searched.searched.AlbumSearchedFragment
import com.project.wave_v2.view.fragment.searched.searched.AllSearchedFragment
import com.project.wave_v2.view.fragment.searched.searched.ArtistSearchedFragment
import com.project.wave_v2.view.fragment.searched.searched.MusicSearchedFragment

class StateFragment(fragment: Fragment) : FragmentStateAdapter(fragment) {
    override fun getItemCount(): Int {
        return 4
    }

    override fun createFragment(position: Int): Fragment {
        var fragment : Fragment
        if (position == 0) {
            fragment = AllSearchedFragment()
        } else if (position == 1) {
            fragment = AlbumSearchedFragment()
        } else if (position == 2) {
            fragment = MusicSearchedFragment()
        } else {
            fragment = ArtistSearchedFragment()
        }

        return fragment
    }

}